lib/common/common/src/mmap_hashmap.rs

```diff
@@ -0,0 +1,233 @@
+use std::collections::{BTreeMap, BTreeSet};
+use std::fs::File;
+use std::io::{self, Cursor, Write as _};
+use std::mem::size_of;
+use std::path::Path;
+use std::str;
+
+use memmap2::Mmap;
+use ph::fmph::Function;
+#[cfg(test)]
+use rand::rngs::StdRng;
+#[cfg(test)]
+use rand::{Rng as _, SeedableRng as _};
+use zerocopy::{AsBytes, FromBytes, FromZeroes};
+
+/// On-disk hash map baked by a memory-mapped file.
+pub struct MmapHashMap {
+    mmap: Mmap,
+    header: Header,
+    phf: Function,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, AsBytes, FromBytes, FromZeroes)]
+struct Header {
+    buckets_pos: u64,
+    buckets_count: u64,
+}
+
+const PADDING_SIZE: usize = 4096;
+
+impl MmapHashMap {
+    /// Save `map` contents to `path`.
+    pub fn create(path: &Path, map: BTreeMap<String, BTreeSet<u32>>) -> io::Result<()> {
```

Comment:
Why `io::Result` instead of our internal `OperationResult`?

Reply:
`OperationResult` is defined in the `segment` crate. So, I've used `io::Result`, just like in the `sparse` crate.

Reply:
got it!